mitosheet/mitosheet/public/v3/sheet_functions/number_functions.py

```diff
@@ -592,6 +592,54 @@ def SUM(*argv: Optional[NumberInputType]) -> NumberFunctionReturnType:
         lambda previous_series, new_series: previous_series + new_series
     )
 
+@cast_values_in_all_args_to_type('number')
+@handle_sheet_function_errors
+def SUMPRODUCT(*argv: Union[pd.Series, pd.DataFrame]) -> NumberFunctionReturnType:
+    """
+    {
+        "function": "SUMPRODUCT",
+        "description": "Returns the sum of the product of the passed arguments.",
+        "search_terms": ["sum product", "sumproduct", "sum", "product", "weighted average"],
+        "examples": [
+            "SUMPRODUCT(A:A, B:B)",
+            "SUMPRODUCT(A:B)"
+        ],
+        "syntax": "SUMPRODUCT(array1, [array2, ...])",
+        "syntax_elements": [{
+                "element": "array1",
+                "description": "The first array argument whose components you want to multiply and then add."
+            },
+            {
+                "element": "value2, ... [OPTIONAL]",
+                "description": "Additional series to multiply then add."
+            }
+        ]
+    }
+    """
+
+    # We need to make sure that all of the passed arguments are the same length
+    # so we can multiply them together. They must be all dataframes or all series
+    for arg in argv:
+        length = len(arg)
+        if length != len(argv[0]):
+            raise ValueError(f"SUMPRODUCT requires all arguments to be the same dimensions.")
```

Comment:
I'm not getting this error, instead I'm getting this: 

<img width="669" alt="Screenshot 2023-05-23 at 4 25 30 PM" src="https://github.com/mito-ds/monorepo/assets/18709905/8f69f9dd-5388-48d7-8f78-a818e42fd644">